### PR TITLE
Whiskers template system

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libdevcore/MiniMoustache.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+///
+/// Moustache-like templates.
+///
+/// Usage:
+///     std::vector<std::map<std::string, std::string>> listValues(2);
+///     listValues[0]["k"] = "key1";
+///     listValues[0]["v"] = "value1";
+///     listValues[1]["k"] = "key2";
+///     listValues[1]["v"] = "value2";
+///     auto s = MiniMoustache("<p1>\n<#list><k> -> <v>\n</list>")
```
Comment (axic) [inline libdevcore/MiniMoustache.h]: https://mustache.github.io/#demo suggests the syntax is different?

**Comment by axic** [comment]
This needs tests.

**Comment by chriseth** [comment]
Sorry, forget to check in the tests.

File: libdevcore/MiniMoustache.h (hunk 1 above)
Comment (chriseth) [reply libdevcore/MiniMoustache.h]: Yes, I replaced `{{` and `}}` by `<` and `>` because those are not part of the syntax of julia.

Comment (axic) [reply libdevcore/MiniMoustache.h]: Gotcha. I guess we should rename it to reflect that it has a different syntax, *mini* to me signals it is a partial implementation, but the syntax has a major differnece :)

Some ideas *Moustachio*, Moustaki, Whiskers.

File: test/libdevcore/MiniMoustache.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+{
+	string templ = "a <#b></b>x";
+	string result = MiniMoustache(templ)("b", vector<MiniMoustache::StringMap>{}).render();
+	BOOST_CHECK_EQUAL(result, "a x");
+}
+
+BOOST_AUTO_TEST_CASE(list)
+{
+	string templ = "a<#b>( <g> - <h> )</b>x";
+	vector<map<string, string>> list(2);
```
Comment (axic) [inline test/libdevcore/MiniMoustache.cpp:86]: `list` is highlighted by Github, does it act as a keyword in some C dialect?

**Comment by axic** [comment]
Added documentation, because ultimately it would be nice having the contributors section populated with dev documentation.

File: libdevcore/MiniMoustache.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	/// Joins the two maps throwing an exception if two keys are equal.
+	static StringMap joinMaps(StringMap const& _a, StringMap const& _b);
+
+	std::string m_template;
+	StringMap m_parameters;
+	StringListMap m_listParameters;
+};
+
+class MiniList
```
Comment (axic) [inline libdevcore/MiniMoustache.h]: This is not used

File: docs/contributing.rst [hunk 4]
```diff
@@ -74,3 +74,13 @@ To run a subset of tests, filters can be used:
 ``soltest -t TestSuite/TestName -- --ipcpath /tmp/testeth/geth.ipc``, where ``TestName`` can be a wildcard ``*``.
 
 Alternatively, there is a testing script at ``scripts/test.sh`` which executes all tests.
+
+Moustachio
```
Comment (axic) [inline docs/contributing.rst]: Obviously this should match the name we decide on.

**Comment by chriseth** [comment]
Sorry, but I HAVE to go with Whiskers if you suggest it.

**Comment by axic** [comment]
@chriseth happy with Whiskers myself.

Didn't know until yesterday that it also refers to human beard :)

File: libdevcore/Whiskers.h [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+	/// Joins the two maps throwing an exception if two keys are equal.
+	static StringMap joinMaps(StringMap const& _a, StringMap const& _b);
+
+	std::string m_template;
+	StringMap m_parameters;
+	StringListMap m_listParameters;
+};
+
+class MiniList
```
Comment (axic) [inline libdevcore/Whiskers.h]: Still present

**Comment by chriseth** [comment]
Updated.

**Review by axic — APPROVED** [review]

**Pull request merged**